pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore.go

```diff
@@ -0,0 +1,453 @@
+//go:build !windows
+// +build !windows
+
+package sigstore
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/hashicorp/go-hclog"
+	"github.com/sigstore/cosign/cmd/cosign/cli/fulcio"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/cosign/bundle"
+	"github.com/sigstore/cosign/pkg/oci"
+	rekor "github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/sigstore/pkg/signature/payload"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// Signature Verification Selector
+	signatureVerifiedSelector = "sigstore-validation:passed"
+)
+
+type Sigstore interface {
+	AttestContainerSignatures(ctx context.Context, status *corev1.ContainerStatus) ([]string, error)
+	FetchImageSignatures(ctx context.Context, imageName string) ([]oci.Signature, error)
+	SelectorValuesFromSignature(oci.Signature, string) (*SelectorsFromSignatures, error)
+	ExtractSelectorsFromSignatures(signatures []oci.Signature, containerID string) []SelectorsFromSignatures
+	ShouldSkipImage(imageID string) (bool, error)
+	AddSkippedImage(imageID []string)
+	ClearSkipList()
+	AddAllowedSubject(subject string)
+	EnableAllowSubjectList(bool)
+	ClearAllowedSubjects()
+	SetRekorURL(rekorURL string) error
+	SetLogger(logger hclog.Logger)
+}
+
+// The following structs are used to go through the payload json objects
+type BundleSignature struct {
+	Content   string            `json:"content"`
+	Format    string            `json:"format"`
+	PublicKey map[string]string `json:"publicKey"`
+}
+
+type BundleSpec struct {
+	Data      map[string]map[string]string `json:"data"`
+	Signature BundleSignature              `json:"signature"`
+}
+
+type BundleBody struct {
+	APIVersion string     `json:"apiVersion"`
+	Kind       string     `json:"kind"`
+	Spec       BundleSpec `json:"spec"`
+}
+
+type SelectorsFromSignatures struct {
+	Subject        string
+	Content        string
+	LogID          string
+	IntegratedTime string
+}
+
+func New(cache Cache, logger hclog.Logger) Sigstore {
+	return &sigstoreImpl{
+		functionHooks: sigstoreFunctionHooks{
+			verifyFunction:             cosign.VerifyImageSignatures,
```

Comment:
Happy to see this being used rather than reimplementing verification logic! A small nit, it was hard to tell that this was being used, it may be worth adding a comment around here noting the importance of `verifyFunction`